### PR TITLE
Address color theming issues

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #253 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Update to use correct colors for theming
* Update dependencies
* Point to auro-library build scripts

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Address color theming issues by updating design tokens and refactoring component registration to use auro-library's runtime utilities. Remove outdated build scripts and update documentation and tests accordingly.

Enhancements:
- Update color theming to use correct design tokens and improve consistency across button variants.
- Refactor component registration to utilize auro-library's runtime utilities for better modularity.

Build:
- Remove obsolete build scripts and update dependencies to align with auro-library standards.

Documentation:
- Update documentation to reflect changes in component registration and theming.

Tests:
- Modify tests to accommodate changes in component registration method.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,13 +1,4 @@
...
- * @param {Object} name - Name to use for custom element.
- * @returns {void}
- */
-export function registerComponent(name) {
-  // alias definition
-  if (!customElements.get(name)) {
-    customElements.define(name, class extends AuroButton {});
-  }
-}
+RuntimeUtils.default.prototype.registerComponent('custom-button', AuroButton);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply index.js:4]: Will keep this in mind for the future.

File: test/auro-button.test.js [hunk 2]
```diff
@@ -33,9 +33,9 @@ describe('auro-button', () => {
   });
 
   it('successfully registers custom component', async () => {
-    registerComponent('test-button');
+    RuntimeUtils.default.prototype.registerComponent('custom-button', AuroButton);
 
-    expect(typeof customElements.get('test-button')).to.equal(typeof AuroButton);
+    expect(typeof customElements.get('custom-button')).to.equal(typeof AuroButton);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-button.test.js:38]: Not editing this test at this time.

File: demo/api.md [hunk 3]
```diff
@@ -563,16 +563,18 @@ The component may be restyled using the following code sample and changing the v
...
-  // THE BELOW TOKEN SEEMS WRONG - WAITING FOR RESPONSE FROM JUNE
-  --ds-auro-button-border-inset-color: var(--ds-color-border-emphasis-inverse, $ds-color-border-emphasis-inverse);
-  --ds-auro-button-loader-color: var(--ds-color-background-darkest, $ds-color-background-darkest);
+  --ds-auro-button-border-color: var(--ds-color-container-ui-primary-default-default, #{$ds-color-container-ui-primary-default-default});
+  --ds-auro-button-container-color: var(--ds-color-container-ui-primary-default-default, #{$ds-color-container-ui-primary-default-default});
+  --ds-auro-button-container-image: var(--ds-color-container-ui-primary-default-default, #{$ds-color-container-ui-primary-default-default});
+  --ds-auro-button-text-color: var(--ds-color-text-primary-inverse, #{$ds-color-text-primary-inverse});
   --ds-auro-button-tap-color: transparent;
+  --ds-auro-button-border-inset-color: var(--ds-color-border-emphasis-inverse, #{$ds-color-border-emphasis-inverse});
+
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/api.md:575]: They do not negatively impact the appearance of the button.

File: docs/partials/index.md [hunk 4]
```diff
@@ -141,8 +141,10 @@ There are two important parts of every Auro component. The <a href="https://deve
 To protect from versioning conflicts with other instances of the component being loaded, it is recommended to use our `registerComponent(name)` method and pass in a unique name.
 
 ```js
-import './node_modules/@aurodesignsystem/auro-button';
-registerComponent('custom-button');
+import { AuroButton } from './src/auro-button.js';
+import * as RuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
+
+RuntimeUtils.default.prototype.registerComponent('custom-button', AuroButton);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply docs/partials/index.md:147]: Not updating at this time.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Screenshot 2024-09-25 at 8 51 09 AM
This text needs to get cleaned up.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**